### PR TITLE
{Profile} az login: Polish UnicodeDecodeError

### LINKED ISSUE #12957: az login fails with UnicodeDecodeError
## Describe the bug

**Command Name**
`az login`

**Errors:**

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-_1mlo87m\azure-cli-core\azure\cli\core\_profile.py", line 1200, in _get_authorization_code_worker
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\socketserver.py", line 453, in __init__
    self.server_bind()
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\http\server.py", line 138, in server_bind
    self.server_name = socket.getfqdn(host)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\socket.py", line 673, in getfqdn
    hostname, aliases, ipaddrs = gethostbyaddr(name)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf8 in position 10: invalid start byte

Login failed
```
My computer has in no way an exotic or esoteric name. It has the humble name "thebeast" ;)

## To Reproduce:

- Ïnstall latest (2.3.1) Azure CLI using MSI (or PowerShell, this fails, too)
- Execute the command: `az login`

## Expected Behavior
A brower opens for enabling me to login.

## Environment Summary
```
Windows-10-10.0.18362-SP0
Python 3.6.6
Installer: MSI
azure-cli 2.3.1

```
## Additional Context
Please note that both
````
az login -u _username_
az login -u _username_ -p _password_
````
successfully logs in.

### PR DESCRIPTION
**Pull request by jiasli**
**Description**  
Fix #12957, #7989
If the computer name or `C:\Windows\System32\drivers\etc\hosts` `127.0.0.1` entries contain international characters (Unicode), Azure CLI fails with a nasty error:

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-_1mlo87m\azure-cli-core\azure\cli\core\_profile.py", line 1200, in _get_authorization_code_worker
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\socketserver.py", line 453, in __init__
    self.server_bind()
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\http\server.py", line 138, in server_bind
    self.server_name = socket.getfqdn(host)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\socket.py", line 673, in getfqdn
    hostname, aliases, ipaddrs = gethostbyaddr(name)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf8 in position 10: invalid start byte

Login failed
```

This PR refines the error message to:

> Please make sure there is no international (Unicode) character in the computer name or C:\Windows\System32\drivers\etc\hosts file's 127.0.0.1 entries. For more details, please see https://github.com/Azure/azure-cli/issues/12957

**Testing Guide**  
Change computer name to a Unicode string like `测试` or add `127.0.0.1 测试` to `C:\Windows\System32\drivers\etc\hosts`. Run `az login`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S168

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -1202,6 +1202,11 @@ def _get_authorization_code_worker(authority_url, resource, results):
             break
         except socket.error as ex:
             logger.warning("Port '%s' is taken with error '%s'. Trying with the next one", port, ex)
+        except UnicodeDecodeError:
+            logger.warning("Please make sure there is no international (Unicode) character in the computer name "
```
Comment (arrownj) [inline src/azure-cli-core/azure/cli/core/_profile.py:1206]: Just curious about that if we actually have international character in the compute name, the only way to get it passed is to rename the compute name ?

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_profile.py:1206]: The link does have instructions about how to bypass it: https://github.com/Azure/azure-cli/issues/12957

But using Unicode as a computer name is always not a good idea.

Comment (arrownj) [reply src/azure-cli-core/azure/cli/core/_profile.py:1206]: Got it.

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 2]
```diff
@@ -1202,6 +1202,11 @@ def _get_authorization_code_worker(authority_url, resource, results):
             break
         except socket.error as ex:
             logger.warning("Port '%s' is taken with error '%s'. Trying with the next one", port, ex)
+        except UnicodeDecodeError:
```
Comment (qianwens) [inline src/azure-cli-core/azure/cli/core/_profile.py:1205]: >except UnicodeDecodeError: [](start = 8, length = 26)

make sure we handle this exception in MSAL too

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_profile.py:1205]: Sure.

**Pull request merged**